crates/nix_health/src/check/direnv.rs

```diff
@@ -31,17 +33,33 @@ impl Checkable for Direnv {
             return checks;
         }
 
-        let direnv_install_check = install_check(self.required);
+        let direnv_install = DirenvInstall::detect();
+
+        let direnv_install_check = install_check(&direnv_install, self.required);
         let direnv_installed = direnv_install_check.result.green();
         checks.push(direnv_install_check);
 
-        if direnv_installed {
+        if !direnv_installed {
+            return checks;
+        }
+
+        let direnv_version = version_check(direnv_install.as_ref().unwrap());
```

Comment:
After moving `direnv_version` and `is_direnv_allowed_on` to impl of `DirenvInstall`, I am passing the `direnv_install` object around, that doesn’t look nice.

Reply:
> I am passing the direnv_install object around, that doesn’t look nice.

That is not a problem in and of itself (structs are meant to be passed around anyway), unless you are referring to something else - such as the resulting code being complex.

I'll take a look, in a bit.

Reply:
> that doesn’t look nice.

I think you are mostly referring to the parts where I added TODO/FIXME in the code. But I want to improve the various `*_check` functions as well.